src/renderer.js

```diff
@@ -124,10 +127,10 @@ export function toArray(text, options = {}) {
     return textWithoutAsciiAliases;
   }
 
-  const textWithouAsciiAliases = replaceAllAsciiAliases(text);
+  const textWithoutAsciiAliases = replaceAllAsciiAliases(text);
 
   return replace(
-    textWithouAsciiAliases.replace(aliasesRegex, replaceAliases),
+    textWithoutAsciiAliases.replace(aliasesRegex, replaceAliases),
```

Comment:
Here we basically want to do this:

1. Replace aliases.
2. Replace ascii aliases to aliases.
3. Replace aliases.

I would do it something like this:

```js
let replacedText = text;
replacedText = replacedText.replace(aliasesRegex, replaceAliases);
replacedText = replaceAllAsciiAliases(replacedText);
replacedText = replacedText.replace(aliasesRegex, replaceAliases);

return replace(replacedText, unicodeEmojiRegex, replaceUnicodeEmoji);
```

Reply:
I was trying to prevent a problem with replaced ascii aliases interfering with the next loop of processing ascii aliases, but it's not working as I intended.

I found an example of a similar problem: `":1<3" -> ":1:heart:" -> "<EMOJI>heart:"`

I suggest changing the algorithm to:

```javascript
let replacedText = text;
replacedText = replacedText.replace(aliasesRegex, replaceAliases);
replacedText = replaceAllAsciiAliases(replacedText);
return replace(replacedText, unicodeEmojiRegex, replaceUnicodeEmoji);
```

and also change `replaceAsciiAliases`:

```javascript
if (!isMaybePartOfBiggerAlias) {
  return aliases[alias];
  // instead of `:${alias}:`
}

return `${aliases[alias]}${maybeBiggerAliasCharacters}`;
// instead of `:${alias}:${maybeBiggerAliasCharacters}`
```

Now we get `":1<3" -> ":1❤️"`.

Reply:
If you agree, I'll make the changes and add tests for those cases.

Reply:
Definitely, that one of the things I meant at https://github.com/tommoor/react-emoji-render/pull/55#issuecomment-626205705. Do whatever you think is required and please add tests for these edge cases that are really good 👍